eventsaver.py

```diff
@@ -0,0 +1,36 @@
+import abc
+
+class EventSaver():
+    __metaclass__ = abc.ABCMeta
+
+    def __init__(self, url):
+        self.url = url
+
+    @abc.abstractmethod
```

Comment:
You already know what the behavior is going to be like.
Provide the default implementation for each of them, instead of repeating it for each concrete class.